File: module_app/src/main/java/com/yunzhu/module/EnvSwitchKit.java
package com.yunzhu.module;

import android.content.Context;

import com.didichuxing.doraemonkit.kit.AbstractKit;
import com.didichuxing.doraemonkit.kit.Category;
import com.yunzhu.module.bus.constant.PrefKey;
import com.yunzhu.module.common.utils.PreferencesUtils;
import com.yunzhu.module.common.utils.ToastUtils;
import com.yunzhu.module.config.AppConfig;

import java.util.UUID;

/**
 * 测试切换设备标识
 * */
public class EnvSwitchKit extends AbstractKit {

    @Override
    public int getCategory() {
        return Category.BIZ;
    }
 
    @Override
    public int getName() {
        return R.string.doraemon_kit_uuid;
    }
 
    @Override
    public int getIcon() {
        return R.drawable.ic_logo;
    }
 
    @Override
    public void onClick(Context context) {
        String uuid = UUID.randomUUID().toString().replace("-","");
        PreferencesUtils.putString(context, PrefKey.DEVICE_UUID,uuid);
        ToastUtils.showShort("修改设备标识为："+uuid);
    }
 
    @Override
    public void onAppInit(Context context) {
    
    }
}

File: module_app/src/main/java/com/yunzhu/module/EnvSwitchKit.kt
package com.yunzhu.module

import android.content.Context
import com.didichuxing.doraemonkit.kit.AbstractKit
import com.didichuxing.doraemonkit.kit.Category
import com.yunzhu.module.bus.constant.PrefKey
import com.yunzhu.module.common.utils.PreferencesUtils
import com.yunzhu.module.common.utils.ToastUtils
import java.util.*

/**
 * 测试切换设备标识
 */
class EnvSwitchKit : AbstractKit() {
    override fun getCategory(): Int {
        return Category.BIZ
    }

    override fun getName(): Int {
        return R.string.doraemon_kit_uuid
    }

    override fun getIcon(): Int {
        return R.drawable.ic_logo
    }

    override fun onClick(context: Context) {
        val uuid = UUID.randomUUID().toString().replace("-", "")
        PreferencesUtils.putString(context, PrefKey.DEVICE_UUID, uuid)
        ToastUtils.showShort("修改设备标识为：$uuid")
    }

    override fun onAppInit(context: Context) {}
}